java/vector/src/main/codegen/templates/UnionMapWriter.java

```diff
@@ -140,6 +140,33 @@ public UnionMapWriter value() {
     return this;
   }
 
+  private String getWriteFieldName() {
+    Field mapField = this.vector.getField();
+    if (mapField == null) {
+      throw new UnsupportedOperationException("MapVector does not have a field.");
+    }
+    if (mapField.getChildren().size() != 1) {
+      throw new UnsupportedOperationException("MapVector does not have a single struct field.");
+    }
+    Field structField = mapField.getChildren().get(0);
```

Comment:
some existing tests failed since structField didn't contain children. So I added a validation. 
Could we do better here?